### PR TITLE
SRCH-125

### PR DESCRIPTION
**Pull request by scott-lewis-nsidc**
Ruby-related fixes (including rubocop and rspec modifications).

### CODE REVIEW

File: spec/suggestions_spec.rb [hunk 1]
```diff
@@ -1,40 +1,4 @@
+# frozen_string_literal: true
+
 require_relative 'spec_helper'
 require_relative '../lib/nsidc_open_search/dataset/search/solr_search_suggest'
```
Comment (juliacollins) [inline spec/suggestions_spec.rb:4]: Looks like all tests were removed. Can this file be deleted? Was the logic moved somewhere else?

File: spec/solr_spec.rb [hunk 2]
```diff
@@ -1,269 +1,318 @@
...
+          hash_including(default_search_expectations),
+          any_args
+        )
+      end
 
-    it 'strips single character words/numbers from phrase strings with a keyword search' do
-      get_should_receive_expectations 'q' => 'sea AND ice'
-      solr_search.execute base_search_parameters.merge(searchTerms: '"a sea 4 ice"')
-    end
+      it 'generats default query with empty keyword search' do
```
Comment (juliacollins) [inline spec/solr_spec.rb:58]: Typo in "generats".

**Review by juliacollins — CHANGES_REQUESTED** [review]
A CHANGELOG entry will need to be added.

File: spec/dataset_solr_spec.rb [hunk 3]
```diff
@@ -38,37 +40,34 @@
...
-        solr_response
-      end
-    end
-
     # This is basically duplicating the normal query test, verifying that we use
     # the same constraints but using a different parser/builder.
 
-    it 'generates default query with empty search' do
-      get_should_receive_expectations default_search_expectations
+    it 'generates default query with empty search2' do
```
[the remark this answers is not in the document]
Comment (scott-lewis-nsidc) [reply spec/dataset_solr_spec.rb:51]: I'll fix this.  I had a copy of this when I was testing the refactoring, and forgot to remove that.

File: spec/enricher_spec.rb [hunk 4]
```diff
@@ -1,28 +1,35 @@
...
-      @obj.class.send :include, NsidcOpenSearch::Enricher
+    before do
+      obj.class.send :include, described_class
     end
 
-    it 'should call enrich entry for each entry' do
-      entry_enricher = double('entry enricher', enrich_entry: nil)
-      @obj.class.send :entry_enrichers, [entry_enricher]
+    it 'calls enrich entry for each entry' do
+      entry_enricher = instance_double(NsidcOpenSearch::EntryEnrichers::Dummy, enrich_entry: nil)
```
[the remark this answers is not in the document]
Comment (scott-lewis-nsidc) [reply spec/enricher_spec.rb:21]: I can just remove them, the only "enricher" that exists is a Dummy one.

File: spec/suggestions_spec.rb (hunk 1 above)
Comment (scott-lewis-nsidc) [reply spec/suggestions_spec.rb:4]: Let me take a look.  The change to this file was an automated one from rubocop, so I overlooked that the file itself was empty.

File: lib/nsidc_open_search/dataset_osdd.rb [hunk 5]
```diff
@@ -14,9 +19,9 @@ def self.new(base_url)
         namespace 'nsidc', 'http://nsidc.org/ns/opensearch/1.1/'
         short_name 'NSIDC Data Set OpenSearch Service'
         long_name 'Data set level OpenSearch interface for nsidc.org'
-        description 'OpenSearch description document that describes how to query our server for'\
-                    'data set level information. This service is currently in incubation; it is'\
-                    'publicly accessible but the interface is not yet stable and custom elements'\
+        description 'OpenSearch description document that describes how to query our server for' \
```
[the remark this answers is not in the document]
Comment (scott-lewis-nsidc) [reply lib/nsidc_open_search/dataset_osdd.rb:22]: Will do!

**Review by juliacollins — APPROVED** [review]

**Pull request merged**